ignite/pkg/cosmosgen/generate_vuex.go

```diff
@@ -0,0 +1,112 @@
+package cosmosgen
+
+import (
+	"github.com/ignite-hq/cli/ignite/pkg/cosmosanalysis/module"
+	"github.com/ignite-hq/cli/ignite/pkg/giturl"
+	"github.com/ignite-hq/cli/ignite/pkg/gomodulepath"
+
+	"os"
+	"path/filepath"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type vuexGenerator struct {
+	g *generator
+}
+
+type generateVuexPayload struct {
+	Modules []module.Module
+	User    string ``
+	Repo    string ``
+}
+
+func newVuexGenerator(g *generator) *vuexGenerator {
+	return &vuexGenerator{
+		g: g,
+	}
+}
+
+func (g *generator) generateVuex() error {
+	vsg := newVuexGenerator(g)
+
+	chainPath, _, err := gomodulepath.Find(g.appPath)
+	if err != nil {
+		return err
+	}
+
+	chainInfo, err := giturl.Parse(chainPath.RawPath)
```

Comment:
We will need some refactoring, same as the comment on `ignite/pkg/cosmosgen/generate_typescript.go`